core/src/main/scala/org/bykn/bosatsu/MemoryMain.scala

```diff
@@ -1,17 +1,18 @@
 package org.bykn.bosatsu
 
 import cats.MonadError
-import cats.data.Kleisli
+import cats.data.{Chain, Kleisli, Validated}
 import com.monovore.decline.Argument
 import scala.collection.immutable.SortedMap
 import org.bykn.bosatsu.tool.Output
 import org.typelevel.paiges.Doc
 
 import cats.syntax.all._
+import cats.data.ValidatedNel
 
-class MemoryMain[G[_], K](
-  platform: PlatformIO[Kleisli[G, MemoryMain.State[K], *], K]) extends
-  MainModule[Kleisli[G, MemoryMain.State[K], *], K](platform) {
+class MemoryMain[G[_]](
+  platform: PlatformIO[Kleisli[G, MemoryMain.State, *], Chain[String]]) extends
+  MainModule[Kleisli[G, MemoryMain.State, *], Chain[String]](platform) {
```

Comment:
next we should change this to being StateT not Kleisli, then we can support writing as well as reading and we can also include a stdout and stderr `Chain[String]` of lines to print.